src/colvarbias_opes.cpp

```diff
@@ -0,0 +1,1977 @@
+  }
+  m_sum_weights = std::pow(m_epsilon, m_bias_prefactor);
+  m_sum_weights2 = m_sum_weights * m_sum_weights;
+  if (m_explore) {
+    get_keyval(conf, "kernelCutoff", m_cutoff, std::sqrt(2.0*m_barrier/m_kbt));
+  } else {
+    get_keyval(conf, "kernelCutoff", m_cutoff, std::sqrt(2.0*m_barrier/m_bias_prefactor/m_kbt));
+  }
+  if (m_cutoff <= 0) {
+    return cvm::error("you must choose a value of kernelCutoff greater than zero");
+  }
+  m_cutoff2 = m_cutoff * m_cutoff;
+  m_val_at_cutoff = std::exp(-0.5 * m_cutoff2);
+  get_keyval(conf, "compressionThreshold", m_compression_threshold, 1);
+  if (m_compression_threshold != 0) {
+    if (m_compression_threshold < 0 || m_compression_threshold > m_cutoff) {
+      return cvm::error("compressionThreshold cannot be smaller than 0 or larger than kernelCutoff", COLVARS_INPUT_ERROR);
+    }
+  }
+  m_compression_threshold2 = m_compression_threshold * m_compression_threshold;
+  get_keyval(conf, "neighborList", m_nlist, false);
+  if (m_nlist) {
+    get_keyval(conf, "neighborListNewHillReset", m_nlist_pace_reset, false);
+    std::vector<cvm::real> nlist_param;
+    get_keyval(conf, "neighborListParameters", nlist_param, std::vector<cvm::real>());
+    if (nlist_param.empty()) {
+      m_nlist_param[0] = 3.0; //*cutoff2_ -> max distance of neighbors
+      m_nlist_param[1] = 0.5; //*nlist_dev2_[i] -> condition for rebuilding
+    } else {
+      if (nlist_param.size() != 2) {
+        return cvm::error("two cutoff parameters are needed for the neighbor list", COLVARS_INPUT_ERROR);
+      }
+      if (nlist_param[0] <= 1.0) {
+        return cvm::error("the first of neighborListParam must be greater than 1.0. The smaller the first, the smaller should be the second as well", COLVARS_INPUT_ERROR);
+      }
+      const cvm::real min_PARAM_1 = (1.-1./std::sqrt(nlist_param[0]))+0.16;
+      if (nlist_param[1] <= 0) {
+        return cvm::error("the second of neighborListParam must be greater than 0", COLVARS_INPUT_ERROR);
+      }
+      if (nlist_param[1] > min_PARAM_1) {
+        return cvm::error("the second of neighborListParam must be smaller to avoid systematic errors. Largest suggested value is: 1.16-1/sqrt(param_0) = " + cvm::to_str(min_PARAM_1), COLVARS_INPUT_ERROR);
+      }
+      m_nlist_param = nlist_param;
+    }
+    m_nlist_center.resize(num_variables());
+    m_nlist_dev2.resize(num_variables(), 0);
+    m_nlist_steps = 0;
+    m_nlist_update = true;
+  }
+  get_keyval(conf, "noZed", m_no_zed, false);
+  if (m_no_zed) {
+    m_sum_weights = 1;
+    m_sum_weights2 = 1;
+  }
+  get_keyval(conf, "fixedGaussianSigma", m_fixed_sigma, false);
+  get_keyval(conf, "recursiveMerge", m_recursive_merge, true);
+  get_keyval(conf, "calcWork", m_calc_work, false);
+  bool b_replicas = false;
+  get_keyval(conf, "multipleReplicas", b_replicas, false);
+  if (!cvm::proxy->b_smp_active) m_num_threads = 1;
+  else m_num_threads = cvm::proxy->smp_num_threads();
+#ifdef OPES_THREADING
```

Comment:
I don't see this environment variable being defined anywhere now.

Reply:
For the time being all threads in Colvars are used for parallelizing the CVCs (the outer loops), so it is impossible to reuse them for summing Gaussian kernels (the inner loops). The code are there but I haven't tested them yet. I plan to enable and test them once you and @giacomofiorin change your mind to use a fine-grained parallelization scheme (parallelizing the inner loops).